editor/src/components/canvas/canvas-strategies/canvas-strategy-types.ts

```diff
@@ -0,0 +1,126 @@
+import type { Spec } from 'immutability-helper'
+import type { CanvasPoint, CanvasVector } from '../../../core/shared/math-utils'
+import type {
+  EditorState,
+  EditorStatePatch,
+  TransientCanvasState,
+  TransientFilesState,
+} from '../../editor/store/editor-state'
+import type { EdgePosition } from '../canvas-types'
+
+type BoundingArea = {
+  type: 'BOUNDING_AREA'
+}
+
+type ResizeHandle = {
+  type: 'RESIZE_HANDLE'
+  edgePosition: EdgePosition
+}
+
+type FlexGapHandle = {
+  type: 'FLEX_GAP_HANDLE'
+}
+
+export interface FlexAlignControlRectProps {
+  x: number
+  y: number
+  width: number
+  height: number
+  highlighted: boolean
+  associatedFlexProp: any // TODO
+}
+
+export interface FlexGapControlRectProps {
+  x: number
+  y: number
+  width: number
+  height: number
+}
+
+type CanvasControlType = BoundingArea | ResizeHandle | FlexGapHandle
+
+export interface CanvasStrategy {
+  name: string
+  updateFn: CanvasStrategyUpdateFn
+  fitnessFn: CanvasStrategyFitnessFn
+}
+
+export type CanvasSessionPatch = Spec<SelectModeCanvasSession>
+
+export type CanvasStrategyUpdateFnResult = {
+  newSessionState: SelectModeCanvasSessionState
+  transientFilesState: TransientFilesState
+  editorStatePatch: EditorStatePatch
+}
+
+export type CanvasStrategyUpdateFn = (
+  lifecycle: 'transient' | 'final',
+  editorState: EditorState,
+  sessionProps: SelectModeCanvasSessionProps,
+  sessionState: SelectModeCanvasSessionState,
+) => CanvasStrategyUpdateFnResult
+
+export type CanvasStrategyFitnessFn = (
+  editorState: EditorState,
+  sessionProps: SelectModeCanvasSessionProps,
+  sessionState: SelectModeCanvasSessionState,
+) => number
+
+export type CanvasInteractionSession = SelectModeCanvasSession
+
+export interface SelectModeCanvasSession {
+  type: 'SELECT_MODE_CANVAS_SESSION'
+  sessionProps: SelectModeCanvasSessionProps
+}
+
+export interface SelectModeCanvasSessionProps {
+  start: CanvasPoint
+  mousePosition: CanvasPoint
+  drag: CanvasVector | null
+  activeControl: CanvasControlType
+  globalTime: number
+  lastTimeMouseMoved: number
+}
+
+export interface SelectModeCanvasSessionState {
+  activeStrategy: CanvasStrategy | null // should this live here? should it exist at all?
```

Comment:
Looks like we're using it for the indicator at least.

```suggestion
  activeStrategy: CanvasStrategy | null
```